Deprecate the overhead circuit breaker setting (#10549)
crate/crate: docs/config/cluster.rst

.. _indices.breaker.query.overhead:

**indices.breaker.query.overhead**
  | *Default:*   ``1.00``
  | *Runtime:*   ``no``

>   This setting is deprecated and changing it has no longer any effect.

Reviewer: might be worth putting this in a CAUTION admonition?

(here and elsewhere)

>   This setting is deprecated and changing it has no longer any effect.

Reviewer: ```suggestion
  .. CAUTION::
  
      This setting is deprecated has no effect.
```
